tensorflow-framework/src/main/java/org/tensorflow/framework/initializers/Initializer.java

```diff
@@ -18,19 +18,16 @@
 import org.tensorflow.types.TInt64;
 import org.tensorflow.types.family.TType;
 
-/**
- * An interface for Initializers
- *
- * @param <T> The data Type for initializer operation
- */
-public interface Initializer<T extends TType> {
+/** An interface for Initializers */
+public interface Initializer {
 
   /**
    * Generates the operation used to perform the initialization.
    *
    * @param dims the shape dimensions
    * @param type the type of tensor
    * @return An operand for the initialization.
+   * @param <T> The data Type for initializer operation
    */
-  Operand<T> call(Operand<TInt64> dims, Class<T> type);
+  <T extends TType> Operand<T> call(Operand<TInt64> dims, Class<T> type);
```

Comment:
We could do that and have the calling program cast it to `TBool` if desired. It only effects `Constant` and `Ones`.

Reply:
I think I would be fine with that. @karllessard @deansher opinions?

Reply:
Makes sense to me.

Going a little further afield: could we / should we make `TBool` a subtype of `TNumber`? When we discuss constraining a tensor type to `TNumber`, we often end up with "yeah, but `TBool`".

Reply:
How about "NotTString" :-)?

Reply:
The earliest versions of Tribuo had `BNumber` which was a boolean box that implemented `java.lang.Number`, but we refactored that out because it was more trouble than it was worth. As such I don't think it's the best idea, especially as `TNumber` lines up fairly well with things you'd put `java.lang.Number` on in Java.

Reply:
Is anyone besides me tempted to add a superclass of `TNumber` and `TBool` titled (perhaps, in a burst of creativity) `TNumberOrBool`?

Reply:
I think that might be nicer for us, but do we want users to have `TNumberOrBool` in their code? Because it looks a little messy.